File: 2022/3/part1.ts

const input = Deno.readTextFileSync("input");

const priority = "abcdefghijklmnopqrstuvwxyzABCDEFGHIJKLMNOPQRSTUVWXYZ".split("");

let sum = 0;
for(const line of input.split("\n")) {
    const firstHalf = line.slice(0, line.length / 2);
    const secondHalf = line.slice(line.length / 2);
    console.log(firstHalf, secondHalf, firstHalf.length, secondHalf.length);

    const sharedLetter = firstHalf.split("").find((letter) => secondHalf.includes(letter));
    sum += priority.indexOf(sharedLetter!) + 1;
    console.log(sharedLetter, priority.indexOf(sharedLetter!) + 1);
}

console.log(sum)

File: 2023/4/part1.py

print(sum([2**(number-1) for number in [
    len(([
        number for number in
        (set(x.strip() for x in (line.split("|")[0].split(":")[1].split(" ")))) & set(x.strip() for x in (line.split("|")[1].split(" "))) if number != ""
    ]))
    for line in open("./input2")
] if number != 0]))